### DISCUSSION TITLE
Cannot run update - Git and Command Line Tools path error

### DISCUSSION THREAD

**Discussion by fabianschulz** [Homebrew/discussions#1128] [Getting started] [answered]
(macOS 11.2.3, Macbook Air M1)

After installing Homebrew according to the instructions everything seems to work OK on my Mac. I was able to install my
packages and they work as expected. However, I cannot update anything and running

`brew update`

throws

```                                                                      
Error: No developer tools installed.
Install the Command Line Tools:
  xcode-select --install
 
Error: 'git' must be installed and in your PATH!
```

Running `brew doctor` throws the same errors. Strangely: during the initial install I have executed the "echo" statements to add Command Line Tools to my path and both Git and Command Line Tools have apparently been installed during the initial setup:

```
~ % which git                                                                                  
/usr/bin/git
~ % xcode-select -p                                                                        
/Library/Developer/CommandLineTools 
```

Did I miss something? Any help would be appreciated!

**Comment by carlocab** [comment]
What's the full output of
```
brew config
brew doctor
ls -l /Library/Developer/CommandLineTools/SDKs
```
?

**Reply by fabianschulz** [reply]
Thanks Carlo!

```
~ % brew config
HOMEBREW_VERSION: >=2.5.0 (shallow or no git repository)
ORIGIN: (none)
HEAD: (none)
Last commit: never
Core tap ORIGIN: (none)
Core tap HEAD: (none)
Core tap last commit: never
Core tap branch: (none)
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: N/A
Git: N/A
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-arm64
CLT: N/A
Xcode: N/A
Rosetta 2: false
```

```
~ % brew doctor    
Warning: Git could not be found in your PATH.
Homebrew uses Git for several internal functions, and some formulae use Git
checkouts instead of stable tarballs. You may want to install Git:
  brew install git
 
Warning: No developer tools installed.
Install the Command Line Tools:
  xcode-select --install
```

There seems to be an issue with the SDKs path ?!

```
~ % ls -l /Library/Developer/CommandLineTools/SDKs
ls: /Library/Developer/CommandLineTools/SDKs: No such file or directory
```

**Reply by carlocab** [reply]
What happens when you run
```
xcode-select --install
```
?

**Comment by SMillerDev** [comment]
Have you tried doing what brew is telling you and running `xcode-select --install`?

**Reply by fabianschulz** [reply]
I was assuming that this was already installed during the setup routine (didn't want to mess up anything before running another install and confirm beforehand). 

noob question, but why do I need to run `xcode-select --install` when `xcode-select -p` shows already "/Library/Developer/CommandLineTools" ?

And will this eventually also "fix" the Git path issue or is this something else I need to look at?

**Reply by carlocab** [reply]
Please run `xcode-select --install`.

**Reply by SMillerDev** [reply]
> I was assuming that this was already installed during the setup routine

It was but recent macOS updates remove it again for no apparent reason

> noob question, but why do I need to run xcode-select --install when xcode-select -p shows already "/Library/Developer/CommandLineTools" ?

Because `-p` only prints the setting, it doesn't validate anything. You can set it to `/not-a/real/path` and `-p`
 will happily print it.

> And will this eventually also "fix" the Git path issue or is this something else I need to look at?

I guess we'll see.

**Comment by fabianschulz** [comment, marked as answer]
Update: `xcode-select --install` has fixed it indeed. Now `brew update` works as expected and `brew doctor` shows no issues.

Thank you both @carlocab and @SMillerDev for helping out! 👍🏻
